File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    console.log("카드 1", cards1)   
   console.log("카드 2", cards2)   
    console.log("목표", goal)   
    
    for(let i = 0; i < goal.length; i++){
        const word = goal[i];
        
        if(cards1[0] === word){
            cards1.shift()
        } else if(cards2[0] === word){
            cards2.shift()
        } else {
            return "No";
        }
    }
    
    return "Yes";
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    cards1Index = 0
    cards2Index = 0
    
    for g in goal:
        if cards1Index < len(cards1) and g == cards1[cards1Index]:
            cards1Index += 1
            continue
        elif cards2Index < len(cards2) and g == cards2[cards2Index]:
            cards2Index += 1
            continue
        else:
            return 'No'
        
    return 'Yes'